constantine/eth_verkle_ipa/barycentric_form.nim

```diff
@@ -38,112 +39,102 @@ func newPrecomputedWeights* [PrecomputedWeights] (res: var PrecomputedWeights) =
-  if element <= uint64(VerkleDomain):
-    var domain_element_Fr: Fr[Banderwagon]
-    var bigndom : matchingOrderBigInt(Banderwagon)
-    bigndom.setUint(uint64(element))
-    domain_element_Fr.fromBig(bigndom)
-
-    res.setOne()
+  doAssert element <= VerkleDomain, "Element should be lesser than the Verkle Domain"
+  var domain_element_Fr {.noInit.} : Fr[Banderwagon]
+  domain_element_Fr.fromInt(element)
 
-    for i in uint64(0) ..< uint64(VerkleDomain):
-      if i == element:
-        continue
+  res.setOne()
 
-      var i_Fr: Fr[Banderwagon] 
+  for i in 0 ..< VerkleDomain:
+    if i == element:
+      continue
 
-      var bigi:  matchingOrderBigInt(Banderwagon)
-      bigi.setUint(uint64(i))
-      i_Fr.fromBig(bigi)
+    var i_Fr {.noInit.} : Fr[Banderwagon] 
+    i_Fr.fromInt(i)
 
-      var temp: Fr[Banderwagon]
-      temp.diff(domain_element_Fr,i_Fr)
-      res.prod(res, temp)
+    var temp {.noInit.} : Fr[Banderwagon]
+    temp.diff(domain_element_Fr,i_Fr)
+    res *= temp
 
 
 func computeBarycentricCoefficients*(res_inv: var openArray[Fr[Banderwagon]], precomp: PrecomputedWeights, point : Fr[Banderwagon]) =
   ## computeBarycentricCoefficients computes the coefficients for a point `z` such that
   ## when we have a polynomial `p` in Lagrange basis, the inner product of `p` and barycentric coefficients is 
   ## equal to p(z). Here `z` is a point outside of the domain. We can also term this as Lagrange Coefficients L_i.
-  var res {.noInit.}: array[VerkleDomain, Fr[Banderwagon]]
+  var res {.noInit.} : array[VerkleDomain, Fr[Banderwagon]]
   for i in 0 ..< VerkleDomain:
-    var weight: Fr[Banderwagon]
+    var weight {.noInit.} : Fr[Banderwagon]
+    var i_fr {.noInit.} : Fr[Banderwagon]
     weight = precomp.barycentricWeights[i]
-    var i_bg: matchingOrderBigInt(Banderwagon)
-    i_bg.setUint(uint64(i))
-    var i_fr: Fr[Banderwagon]
-    i_fr.fromBig(i_bg)
+    i_fr.fromInt(i)
 
     res[i].diff(point, i_fr)
-    res[i].prod(res[i], weight)
+    res[i] *= weight
 
-  var totalProd: Fr[Banderwagon]
+  var totalProd {.noInit.} : Fr[Banderwagon]
   totalProd.setOne()
 
   for i in 0 ..< VerkleDomain:
-    var i_bg: matchingOrderBigInt(Banderwagon)
-    i_bg.setUint(uint64(i))
-    var i_fr: Fr[Banderwagon]
-    i_fr.fromBig(i_bg)
+    var i_fr {.noInit.} : Fr[Banderwagon]
+    i_fr.fromInt(i)
 
-    var tmp: Fr[Banderwagon]
+    var tmp {.noInit.} : Fr[Banderwagon]
     tmp.diff(point, i_fr)
 
-    totalProd.prod(totalProd, tmp)
+    totalProd *= tmp
 
   res_inv.batchInvert(res)
 
   for i in 0 ..< VerkleDomain:
-    res_inv[i].prod(res_inv[i], totalprod)
+    res_inv[i] *= totalProd
 
 
 func getInvertedElement*(res: var Fr[Banderwagon], precomp : PrecomputedWeights, element : int, is_negative: bool) =
   var index: int
   index = element - 1 
 
   if is_negative:
-    var midpoint = int((len(precomp.invertedDomain) / 2)) - 1
+    var midpoint = int((len(precomp.invertedDomain) / 2))
     index = index + midpoint
```

Comment:
Is -1/0 properly handled? This needs a comment mentioning this gotcha and ideally a test.

Reply:
It should be, for I'm not including 1/0 and -1/0 cases while populating the inverted domain array. Yet I can write a very specific test for this in the next pass, for ipa and multiproof tests.